### PR TITLE
improve: send impression events to a custom Analytics account (SDKCF-5252)

### PR DESCRIPTION
**Pull request by imaginaris**
# Description
~* Impression event are now sent to a dedicated Analytics account~
~* The account number can be set in Info.plist; If it's not set, a default account will be used~
* IAM user events logging to Analytics has been removed
* Display impression event is now sent to Analytics separately

## Links
SDKCF-5252

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [X] I ran `fastlane ci` without errors

### CODE REVIEW

File: Podfile
Comment (imaginaris) [inline Podfile]: The improvement relies on the unreleased changes in RSDKUtils

File: Sources/RInAppMessaging/Utilities/BundleInfo.swift
Comment (imaginaris) [inline Sources/RInAppMessaging/Utilities/BundleInfo.swift]: `1` is a temporary value. It will be updated before merge

File: Podfile
Comment (imaginaris) [reply Podfile]: To be honest I'm not sure what's the best approach. Update Podspec and Package.swift to use master branch version of RSDKUtils, wait for RSDKUtils release, or temporarily bring new SDKUtils changes into IAM SDK?

Comment (malayli) [reply Podfile]: @imaginaris I think it is better to keep RSDKUtils changes in IAM SDK until the release of RSDKUtils.
When will RSDKUtils be released?

Comment (imaginaris) [reply Podfile]: @malayli There's no plan for release yet

Comment (malayli) [reply Podfile]: I see. I think it is preferable to keep `pod 'RSDKUtils', '~> 3.0.0'` in the IAM SDK. @imaginaris 
You can create a ticket to move the code to RSDKUtils in the technical debt doc.

File: Sources/RInAppMessaging/Utilities/BundleInfo.swift
Comment (malayli) [inline Sources/RInAppMessaging/Utilities/BundleInfo.swift]: Does it need to be `class var`? Is it for overriding `bundle`? or can it be `static var`?

File: Podfile
Comment (imaginaris) [reply Podfile]: @malayli Done. (SDKCF-5284)

File: Sources/RInAppMessaging/Utilities/BundleInfo.swift
Comment (imaginaris) [reply Sources/RInAppMessaging/Utilities/BundleInfo.swift]: Yes, I used `class` instead of `static` for overriding/mocking in unit tests

Comment (malayli) [reply Sources/RInAppMessaging/Utilities/BundleInfo.swift]: Ok 👍

**Review by malayli — APPROVED** [review]

**Review by donnie-jp — CHANGES_REQUESTED** [review]
The strategy for this has changed. Please sync with PDM but I don't think we will need this change.

**Comment by imaginaris** [comment]
> The strategy for this has changed. Please sync with PDM but I don't think we will need this change.

Updated the PR based on PDM's decision.

**Review by donnie-jp — APPROVED** [review]
LGTM

**Pull request merged**
